### PR TITLE
Fix allow storage constraints to search both physical and virtual(logi…

### PR DESCRIPTION
**Pull request by ExternalReality**
…cal) block devices.

### CODE REVIEW

File: interfaces.go [hunk 1]
```diff
@@ -223,6 +223,9 @@ type Machine interface {
 
 	// BlockDevices returns all the physical and virtual block devices on the machine.
 	BlockDevices() []BlockDevice
+	// PhysicalBlockDevice returns the physical block device for the machine
```
Comment (jameinel) [inline interfaces.go]: s/PhysicalBlockDevice/BlockDevice/

and probably not "the physical block device"

**Review by jameinel — COMMENTED** [review]
If its just exposing a different field (BlockDevice) rather than (PhysicalBlockDevice), that seems good.
Do we know that all entries that we've been reading from PhysicalBlockDevices are present in BlockDevices?
If so, LGTM (caveat some comment tweaks, and we should at least have 1 canned test that grabs a MAAS output for Block devices and mocks the response.)

**Review by anastasiamac — APPROVED** [review]
Looks really neat!
Thank you for clear explanatory comments :)

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**